app/scripts/modules/core/src/domain/INotificationTypeConfig.ts

```diff
@@ -1,11 +1,11 @@
 import React from 'react';
-import { INotificationSettings } from 'core/config';
 
 export interface INotificationTypeConfig {
   label: string;
-  key: keyof INotificationSettings;
+  key: string;
   config?: INotificationTypeCustomConfig;
   component?: React.ComponentType;
+  isExtensionNotification?: boolean;
```

Comment:
This moves knowledge of plugins *into* the registry, which doesn't feel quite right to me. What other options had you considered?

Reply:
I wanted to avoid making users have to configure the plugin AND opt in to the notification in `SETTINGS.notifications`. But if that had to happen, that's OK.

Another option, which is definitely better than what I wrote, would be to include an optional `enabled` field on `INotificationTypeConfig`. This would effectively be the same thing but less plugin-specific.

Reply:
I think we could move the registration of the built-in notifications into a separate function.  That way, the registry itself doesn't check if something is `.enabled`. 

For example:
```diff
import { Registry } from 'core/registry';

import { BearychatNoficationType } from './BearychatNotificationType';

-Registry.pipeline.registerNotification({
+export const bearyChatNotification = {
  component: BearychatNoficationType,
  key: 'bearychat',
  label: 'Bearychat',
};
```

Then, somewhere else like `notification/notification.types.ts` you could do:

psuedocode
```js
import { bearyChatNotification } from './types'
import { slackNotification } from './types'
...

const builtInNotificationTypes = [bearyChatNotification, slackNotification, ...other types].forEach(type =>  {
  if (SETTINGS. /* check settings.enabled for the built in ones */) {
    Registry.registerTheEnabledNotification(type);
  }
});
```

Reply:
done